### PR TITLE
[SPARK-30889][CORE][DOC] Add version information to the configuration of Worker

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
1.Add version information to the configuration of `Worker`.
2.Update the docs of `Worker`.

I sorted out some information show below.

Item name | Since version | JIRA ID | Commit ID | Note
-- | -- | -- | -- | --
spark.worker.resourcesFile | 3.0.0 | SPARK-27369 | 7cbe01e8efc3f6cd3a0cac4bcfadea8fcc74a955#diff-b2fc8d6ab7ac5735085e2d6cfacb95da |  
spark.worker.timeout | 0.6.2 | None | e395aa295aeec6767df798bf1002b1f30983c1cd#diff-776a630ac2b2ec5fe85c07ca20a58fc0 |  
spark.worker.driverTerminateTimeout | 2.1.2 | SPARK-20843 | ebd72f453aa0b4f68760d28b3e93e6dd33856659#diff-829a8674171f92acd61007bedb1bfa4f |  
spark.worker.cleanup.enabled | 1.0.0 | SPARK-1154 | 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.cleanup.interval | 1.0.0 | SPARK-1154 | 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.cleanup.appDataTtl | 1.0.0 | SPARK-1154 | 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.preferConfiguredMasterAddress | 2.2.1 | SPARK-20529 | 75e5ea294c15ecfb7366ae15dce196aa92c87ca4#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.ui.port | 1.1.0 | SPARK-2857 | 12f99cf5f88faf94d9dbfe85cb72d0010a3a25ac#diff-48ca297b6536cb92362bec1487581f05 |  
spark.worker.ui.retainedExecutors | 1.5.0 | SPARK-9202 | c0686668ae6a92b6bb4801a55c3b78aedbee816a#diff-916ca56b663f178f302c265b7ef38499 | 
spark.worker.ui.retainedDrivers | 1.5.0 | SPARK-9202 | c0686668ae6a92b6bb4801a55c3b78aedbee816a#diff-916ca56b663f178f302c265b7ef38499 | 
spark.worker.ui.compressedLogFileLengthCacheSize | 2.0.2 | SPARK-17711 | 26e978a93f029e1a1b5c7524d0b52c8141b70997#diff-d239aee594001f8391676e1047a0381e |  
spark.worker.decommission.enabled | 3.1.0 | SPARK-20628 | d273a2bb0fac452a97f5670edd69d3e452e3e57e#diff-b2fc8d6ab7ac5735085e2d6cfacb95da |  

### Why are the changes needed?
Supplemental configuration version information.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Exists UT

### CODE REVIEW

**Comment by beliefer** [comment]
retest this please.

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 1]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
     .doc("Path to a file containing the resources allocated to the worker. " +
       "The file should be formatted as a JSON array of ResourceAllocation objects. " +
       "Only used internally in standalone mode.")
+    .version("3.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-27369, commit ID: 7cbe01e8efc3f6cd3a0cac4bcfadea8fcc74a955#diff-b2fc8d6ab7ac5735085e2d6cfacb95da

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 2]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .stringConf
     .createOptional
 
   val WORKER_TIMEOUT = ConfigBuilder("spark.worker.timeout")
+    .version("0.6.2")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: No JIRA ID, commit ID: e395aa295aeec6767df798bf1002b1f30983c1cd#diff-776a630ac2b2ec5fe85c07ca20a58fc0

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 3]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .longConf
     .createWithDefault(60)
 
   val WORKER_DRIVER_TERMINATE_TIMEOUT = ConfigBuilder("spark.worker.driverTerminateTimeout")
+    .version("2.1.2")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-20843, commit ID: ebd72f453aa0b4f68760d28b3e93e6dd33856659#diff-829a8674171f92acd61007bedb1bfa4f

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 4]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .timeConf(TimeUnit.MILLISECONDS)
     .createWithDefaultString("10s")
 
   val WORKER_CLEANUP_ENABLED = ConfigBuilder("spark.worker.cleanup.enabled")
+    .version("1.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-1154, commit ID: 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 5]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .booleanConf
     .createWithDefault(false)
 
   val WORKER_CLEANUP_INTERVAL = ConfigBuilder("spark.worker.cleanup.interval")
+    .version("1.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-1154, commit ID: 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 6]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .longConf
     .createWithDefault(60 * 30)
 
   val APP_DATA_RETENTION = ConfigBuilder("spark.worker.cleanup.appDataTtl")
+    .version("1.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-1154, commit ID: 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 7]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .longConf
     .createWithDefault(7 * 24 * 3600)
 
   val PREFER_CONFIGURED_MASTER_ADDRESS = ConfigBuilder("spark.worker.preferConfiguredMasterAddress")
+    .version("2.2.1")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-20529, commit ID: 75e5ea294c15ecfb7366ae15dce196aa92c87ca4#diff-916ca56b663f178f302c265b7ef38499

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 8]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .booleanConf
     .createWithDefault(false)
 
   val WORKER_UI_PORT = ConfigBuilder("spark.worker.ui.port")
+    .version("1.1.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-2857, commit ID: 12f99cf5f88faf94d9dbfe85cb72d0010a3a25ac#diff-48ca297b6536cb92362bec1487581f05

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 9]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .intConf
     .createOptional
 
   val WORKER_UI_RETAINED_EXECUTORS = ConfigBuilder("spark.worker.ui.retainedExecutors")
+    .version("1.5.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-9202, commit ID: c0686668ae6a92b6bb4801a55c3b78aedbee816a#diff-916ca56b663f178f302c265b7ef38499

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 10]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .intConf
     .createWithDefault(1000)
 
   val WORKER_UI_RETAINED_DRIVERS = ConfigBuilder("spark.worker.ui.retainedDrivers")
+    .version("1.5.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-9202, commit ID: c0686668ae6a92b6bb4801a55c3b78aedbee816a#diff-916ca56b663f178f302c265b7ef38499

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 11]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
     .intConf
     .createWithDefault(1000)
 
   val UNCOMPRESSED_LOG_FILE_LENGTH_CACHE_SIZE_CONF =
     ConfigBuilder("spark.worker.ui.compressedLogFileLengthCacheSize")
-    .intConf
-    .createWithDefault(100)
+      .version("2.0.2")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-17711, commit ID: 26e978a93f029e1a1b5c7524d0b52c8141b70997#diff-d239aee594001f8391676e1047a0381e

File: core/src/main/scala/org/apache/spark/internal/config/Worker.scala [hunk 12]
```diff
@@ -28,52 +28,64 @@ private[spark] object Worker {
...
+      .intConf
+      .createWithDefault(100)
 
   private[spark] val WORKER_DECOMMISSION_ENABLED =
     ConfigBuilder("spark.worker.decommission.enabled")
+      .version("3.1.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Worker.scala]: SPARK-20628, commit ID: d273a2bb0fac452a97f5670edd69d3e452e3e57e#diff-b2fc8d6ab7ac5735085e2d6cfacb95da

**Review by HyukjinKwon — APPROVED** [review]

**Pull request closed without merging**